pkg/cloud/vsphere/actuators/machine/actuator.go

```diff
@@ -302,5 +312,82 @@ func (a *Actuator) reconcileReadyState(ctx *context.MachineContext) error {
 		ctx.Logger.V(6).Info("requeuing until noderef is set")
 		return &clusterErr.RequeueAfterError{RequeueAfter: constants.DefaultRequeue}
 	}
+	// A machine being ready also needs to assign itself an annotation in order
+	// to survive the pivot. Otherwise a race condition exists where the NodeRef
+	// on machines aren't yet set post-pivot, and a new machine may try to
+	// initialize the cluster rather than join it.
+	if ctx.Machine.Annotations == nil {
+		ctx.Machine.Annotations = map[string]string{}
+	}
+	if _, ok := ctx.Machine.Annotations[constants.ReadyAnnotationLabel]; !ok {
+		ctx.Machine.Annotations[constants.ReadyAnnotationLabel] = ""
+		ctx.Logger.V(6).Info("machine is ready")
```

Comment:
Should we log this if the annotation already exists as well?

Reply:
I wanted to reduce the logging, otherwise this appears on every update call.